File: LeetCode/BFS/965. Univalued Binary Tree/Ingyu.java
import javax.swing.tree.TreeNode;
import java.util.LinkedList;
import java.util.Queue;

class Solution {
    public boolean isUnivalTree(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(root);

        int value = queue.peek().val;
        while(!queue.isEmpty()) {
            TreeNode node = queue.poll();
            if (node.val != value) return false;

            if (node.left != null) queue.offer(node.left);
            if (node.right != null) queue.offer(node.right);
        }

        return true;
    }
}

File: LeetCode/BFS/965. Univalued Binary Tree/Ingyu.kt
import java.util.*

class Ingyu {
    fun isUnivalTree(root: TreeNode?): Boolean {
        val queue: Queue<TreeNode> = LinkedList<TreeNode>()
        queue.offer(root)
        val value = queue.peek().`val`

        while (!queue.isEmpty()) {
            val node = queue.poll()

            if (node.`val` != value) return false

            node.left?.let {
                queue.offer(node.left)
            }
            node.right?.let {
                queue.offer(node.right)
            }
        }
        return true
    }
}